lib/baldr/validator.rb

```diff
@@ -24,10 +26,11 @@ def validate_tree!(segment, grammar, structure)
     l = 0
     structure.fetch(:level, []).each do |s|
       loop = segment.children[l]
+
       if loop && loop.id.to_s == s[:id]
-        check_loop_count(loop, s)
+        check_loop_count(loop, s, @truncate_loops)
```

Comment:
@npersad84 _a that doesn't make sense, you're not modifying the hash. you can't use instance variables here anyway because it's a module. you have to pass the entire options hash each time